force-app/main/default/lwc/lookup/lookup.js

```diff
@@ -0,0 +1,422 @@
+    }
+
+    handleNewRecordClick(event) {
+        const objectApiName = event.currentTarget.dataset.sobject;
+        const selection = this.newRecordOptions.find((option) => option.value === objectApiName);
+
+        const preNavigateCallback = selection.preNavigateCallback
+            ? selection.preNavigateCallback
+            : () => Promise.resolve();
+        preNavigateCallback(selection).then(() => {
+            this[NavigationMixin.Navigate]({
+                type: 'standard__objectPage',
+                attributes: {
+                    objectApiName,
+                    actionName: 'new'
+                },
+                state: {
+                    defaultFieldValues: selection.defaults
+                }
+            });
+        });
+    }
+
+    // STYLE EXPRESSIONS
+
+    get hasResults() {
+        return this._searchResults.length > 0;
+    }
+
+    get getFormElementClass() {
+        return this.variant === VARIANT_LABEL_INLINE
+            ? 'slds-form-element slds-form-element_horizontal'
+            : 'slds-form-element';
+    }
+
+    get getLabelClass() {
+        return this.variant === VARIANT_LABEL_HIDDEN
+            ? 'slds-form-element__label slds-assistive-text'
+            : 'slds-form-element__label';
+    }
+
+    get getContainerClass() {
+        let css = 'slds-combobox_container slds-has-inline-listbox ';
+        if (this._hasFocus && this.hasResults) {
+            css += 'slds-has-input-focus ';
+        }
+        if (this.errors.length > 0) {
+            css += 'has-custom-error';
+        }
+        return css;
+    }
+
+    get getDropdownClass() {
+        let css = 'slds-combobox slds-dropdown-trigger slds-dropdown-trigger_click ';
+        const isSearchTermValid = this._cleanSearchTerm && this._cleanSearchTerm.length >= this.minSearchTermLength;
+        if (this._hasFocus && this.isSelectionAllowed() && (isSearchTermValid || this.hasResults)) {
+            css += 'slds-is-open';
+        }
+        return css;
+    }
+
+    get getInputClass() {
+        let css = 'slds-input slds-combobox__input has-custom-height ';
+        if (this.errors.length > 0 || (this._isDirty && this.required && !this.hasSelection())) {
+            css += 'has-custom-error ';
+        }
+        if (!this.isMultiEntry) {
+            css += 'slds-combobox__input-value ' + (this.hasSelection() ? 'has-custom-border' : '');
+        }
+        return css;
+    }
+
+    get getComboboxClass() {
+        let css = 'slds-combobox__form-element slds-input-has-icon ';
+        if (this.isMultiEntry) {
+            css += 'slds-input-has-icon_right';
+        } else {
+            css += this.hasSelection() ? 'slds-input-has-icon_left-right' : 'slds-input-has-icon_right';
+        }
+        return css;
+    }
+
+    get getSearchIconClass() {
+        let css = 'slds-input__icon slds-input__icon_right ';
+        if (!this.isMultiEntry) {
+            css += this.hasSelection() ? 'slds-hide' : '';
```

Comment:
Not that it matters a whole lot, but couldn't this just be:

```suggestion
        if (!this.isMultiEntry && this.hasSelection()) {
            css += 'slds-hide';
```